### PR TITLE
sidecar: Loads the certificate and key when the sidecar starts.

### PR DESCRIPTION
**Pull request by maheshbaliga**
* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

The sidecar did not validate the certificate and key while starting; Reading and parsing of the certificate were done only when the client initiated the TLS handshake with the sidecar. This is when the certificate related issues surfaced. 

With this code change, these files are now loaded and checked by the sidecar when it starts. 

## Verification

Tested the fix by adding unit tests. Also, by running the sidecar and query components with TLS configurations.

### CODE REVIEW

File: pkg/tls/options.go [hunk 1]
```diff
@@ -17,8 +17,8 @@ import (
 )
 
 // NewServerConfig provides new server TLS configuration.
-func NewServerConfig(logger log.Logger, cert, key, clientCA string) (*tls.Config, error) {
-	if key == "" && cert == "" {
+func NewServerConfig(logger log.Logger, certPath, keyPath, clientCA string) (*tls.Config, error) {
+	if keyPath == "" && certPath == "" {
```
Comment (ApsTomar) [inline pkg/tls/options.go:21]: Any specific reasons for changing their var names?

Comment (maheshbaliga) [reply pkg/tls/options.go:21]: I have created a new var `certificate`, the existing name of the certificate path was `cert`, which i thought will cause some confusion. Hence changed it.

**Review by ApsTomar — APPROVED** [review]

**Review by theluckiesthuman — APPROVED** [review]

**Comment by samkulkarni20** [comment]
@maheshbaliga Now that you've changed the server config definition, won't you need to change all the calls to this func and/or update the documentation to provide the cert & key path, in the thanos deployment configurations?
Also, where are the changes which change the cert validation time from TLS call time to container start time?

**Comment by maheshbaliga** [comment]
@samkulkarni20 No, the definition has not changed, only `cert` and `key` variables were renamed to avoid naming confusion. The cert & key modification times are set in `getCertificate` func during the handshake, this code was already present.

**Review by samkulkarni20 — APPROVED** [review]
LGTM

**Pull request closed without merging**